src/app/features/courses/edit-page/edit-page.component.ts

```diff
@@ -0,0 +1,110 @@
+import {Component, OnInit} from '@angular/core';
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from "@angular/forms";
+import {ActivatedRoute, Router} from "@angular/router";
+import {CoursesStoreService} from "../../../services/courses-store.service";
+import {Course} from "../../../services/course";
+import {CoursesService} from "../../../services/courses.service";
+
+@Component({
+  selector: 'app-edit-page',
+  templateUrl: './edit-page.component.html',
+  styleUrls: ['./edit-page.component.css']
+})
+export class EditPageComponent implements OnInit {
+  isEditMode: boolean = false;
+  courseId: string | null = '';
+  public formData = {
+    title: '',
+    description: '',
+    authorName: '',
+    authors: [],
+    duration: 0,
+  }
+
+  editForm: FormGroup;
+
+  constructor(private fb: FormBuilder, private route: ActivatedRoute, private coursesService: CoursesService, private router: Router) {
+    this.editForm = new FormGroup({
+      title: new FormControl(this.formData.title, [
+        Validators.required,
+        Validators.minLength(4)
+      ]),
+      description: new FormControl(this.formData.description, [
+        Validators.required,
+        Validators.minLength(10)
+      ]),
+      authorName: new FormControl(this.formData.authorName, [
+        Validators.required,
+        Validators.minLength(4)
+      ]),
+      authors: new FormArray([]),
+      duration: new FormControl(this.formData.duration, [
+        Validators.required,
+        Validators.min(0)
+      ]),
+    })
+  }
+
+  ngOnInit(): void {
+    this.courseId = this.route.snapshot.paramMap.get('id');
+
+    if (this.courseId) {
+      this.isEditMode = true;
+      this.coursesService.getCourse(this.courseId).subscribe(data => {
+        this.title?.setValue(data.title);
+        this.description?.setValue(data.description);
+        this.duration?.setValue(data.duration);
+        this.authors?.setValue(data.authors);
+      })
```

Comment:
You can try patch method of formGroup https://angular.io/api/forms/FormGroup#patchvalue